src/natcap/invest/utils.py

```diff
@@ -795,3 +795,65 @@ def _assert_vectors_equal(
         expected_vector = None
 
     return None
+
+def _reclassify_raster_op(
+        raster_path_band, value_map, target_raster_path, target_datatype,
+        target_nodata, error_details=None, values_required=True):
+    """A wrapper function for calling ``pygeoprocessing.reclassify_raster``.
+
+    This wrapper function is helpful when added as a ``TaskGraph.task`` so
+    a better error message can be provided to the users if an exception
+    occurs.
+
+    Args:
+        raster_path_band (tuple): a tuple including file path to a raster
+            and the band index to operate over. ex: (path, band_index)
+        value_map (dictionary): a dictionary of values of
+            {source_value: dest_value, ...} where source_value's type is the
+            same as the values in ``base_raster_path`` at band ``band_index``.
+            Must contain at least one value.
+        target_raster_path (string): target raster output path; overwritten if
+            it exists
+        target_datatype (gdal type): the numerical type for the target raster
+        target_nodata (numerical type): the nodata value for the target raster
+            Must be the same type as target_datatype
+        error_details (dict): a dictionary with key value pairs that provide 
+            more context for a raised
+            ``pygeoprocessing.ReclassificationMissingValuesError``.
+            keys must be {'raster_name', 'column_name', 'table_name'}. Values 
+            each key represent:
+                'raster_name' - string for the raster name being reclassified
+                'column_name' - column name from table with which ``value_map``
```

Comment:
Hmm, by `two columns` are you referring to the table column and the raster column? Maybe `table_column_name` would be more explicit about this key?

Reply:
Yeah, I guess I was thinking of `'code'` (or whichever column represents the landcover code) and also the target column, the one containing the reclassification value (`'LITTER'` or what have you).  I'm not sure what a clearer key here might be, but `table_column_name` does sound like a good improvement!

Reply:
@phargogh : Hmmm, yea it definitely is hard to articulate. Maybe I'm not being clear and maybe I'm now just understanding the "two column" bit mentioned earlier. If we have a table with columns: "lucode" and "C_above" then `error_details['column_name']` SHOULD be "lucode". Again, `keys` is confusing, because is it table key or dictionary key... here in `'column_name' - column name from table with which ``value_map`` keys came from` it's dictionary key... 

Thoughts?

Maybe: `'column_name' - column name from table with which ``value_map`` dictionary keys came from.

And for the actual error message, maybe `key` should be replaced with `corresponding`:
```python
                "Values in the{formatted_details['raster_name']}raster were"
                " found that are not represented under the CORRESPONDING"
                f"{formatted_details['column_name']} column of the"
                f"{formatted_details['table_name']}table. The missing values"
                f" found in the{formatted_details['raster_name']}raster but"
                f" not the table are: {err.missing_values}.")
```

Reply:
How about a slight tweak to the docstring you mentioned?

> `'column_name'` - name of the table column that `value_map` dictionary keys came from.

`Corresponding` makes sense to me!